src/components/DatePicker/DatePicker.style.ts

```diff
@@ -57,17 +57,22 @@ export const datePickerStyles = ({ isRangePicker }: { isRangePicker?: boolean })
   .DayPicker-Day {
     padding: 0.7em;
     font-size: ${theme.typography[14]};
+    border-radius: 0;
   }
   .DayPicker-Day--today {
     color: initial;
   }
 
-  .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside),
+  .DayPicker-Day--selected:not(.DayPicker-Day--outside),
   .DayPicker-Day--start:not(.DayPicker-Day--outside) {
     background-color: #dfdfdf;
     color: #000;
   }
 
+  .DayPicker-Day--selected.DayPicker-Day--disabled:not(.DayPicker-Day--outside) {
+    opacity: 0.5;
+  }
+
   .DayPicker-Day:hover {
```

Comment:
I would suggest to update the css selector here as well from `.DayPicker-Dat:hover` to `.DayPicker-Day:not(.DayPicker-Day--disabled):hover`.
This way we can avoid giving the user a false sense of selecting a date that is disabled.
Also, I see that when selecting a day at the moment, the default outline is shown... I think it should be best if we disabled this as well by adding an extra rule, like:
```
.DayPicker-Day:focus {
  outline: none;
}
```

but if we want to keep the outline, since when an active day is selected the selection table collapses and it doesn't affect the UX so much, I would suggest disabling it at least for disabled dates. 😄

Reply:
i agree for both comments